.github/workflows/push-important-models.yml

```diff
@@ -0,0 +1,89 @@
+name: Slow tests on important models (on Push - A10)
+
+on:
+  push:
+    branches: [ main ]
+
+env:
+  RUN_SLOW: "yes"
+  IS_GITHUB_CI: "1"
+  OUTPUT_SLACK_CHANNEL_ID: "C06L2SGMEEA"
+
+jobs:
+  get_modified_models:
+    name: "Get all modified files"
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@3f54ebb830831fc121d3263c1857cfbdc310cdb9 #v42
+        with:
+          files: src/transformers/models/**
+      - name: Run step if only the files listed above change
+        if: steps.changed-files.outputs.any_changed == 'true'
+        id: set-matrix
+        run: |
+            model_arrays=()
+            for file in ${{ steps.changed-files.outputs.all_changed_files }}; do
+                model_path="${file#*models/}"
+                model_path="models/${model_path%%/*}"
+                if grep -qFx "$model_path" utils/important_models.txt; then
+                    echo "$model_path was changed"
+                    # Append the file to the matrix string
+                    model_arrays+=("$model_path")
+                fi
+            done
+            matrix_string=$(printf '"%s", ' "${model_arrays[@]}" | sed 's/, $//')
+            echo "matrix=[$matrix_string]" >> $GITHUB_OUTPUT
+  test_modified_files:
+    needs: get_modified_models
+    name: Run slow tests on modified models
+    runs-on: [single-gpu, nvidia-gpu, a10, ci] # TODO: uncomment this
+    container:
+      image: huggingface/transformers-all-latest-gpu
```

Comment:
think we should use a more lightweight !

Reply:
That docker image has been greatly reduced in https://github.com/huggingface/transformers/pull/29046 - I think it's fine to keep it for now but maybe work on that as a separate PR